syslog/syslog.go

```diff
@@ -90,25 +90,20 @@ type Logger struct {
 
 // Dial connects to the syslog server at raddr, using the optional certBundle,
 // and launches a goroutine to watch logger.Packets for messages to log.
-func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger, error) {
-	// dial once, just to make sure the network is working
-	conn, err := dial(network, raddr, rootCAs)
-
-	if err != nil {
-		return nil, err
-	} else {
-		logger := &Logger{
-			ClientHostname: clientHostname,
-			network:        network,
-			raddr:          raddr,
-			rootCAs:        rootCAs,
-			Packets:        make(chan Packet, 100),
-			Errors:         make(chan error, 0),
-			conn:           conn,
-		}
-		go logger.writeLoop()
-		return logger, nil
+func Dial(clientHostname, network, raddr string, rootCAs *x509.CertPool) (*Logger) {
+	logger := &Logger{
+		ClientHostname: clientHostname,
+		network:        network,
+		raddr:          raddr,
+		rootCAs:        rootCAs,
+		Packets:        make(chan Packet, 100),
+		Errors:         make(chan error, 0),
 	}
+
+	logger.connect()
```

Comment:
This is going to block indefinitely until a successful connection is made, which means that [this logging](https://github.com/papertrail/remote_syslog2/blob/retry-first-connection/remote_syslog.go#L120) will never happen for errors during the initial connection.

I would suggest moving this back to a single `dial()` call above creating the `Logger` instance which will give one chance to connect, but then return an instance of the `Logger` which will allow the debug log to report what is going on.